cloudinit/sources/DataSourceAzure.py

```diff
@@ -1091,6 +1103,11 @@ def _report_ready_for_pps(
                 )
                 report_diagnostic_event(msg, logger_func=LOG.error)
                 raise sources.InvalidMetaDataException(msg) from error
+        except Exception as e:
```

Comment:
> The old behavior logged at error level.

The old one logged at `ERROR` level, but it also caught the top-level exception and then did type checking of that exception type, which is a smell. This is more of a clean-up change than anything, and I think that adding the else clause under here would actually be a change in behavior (before it would only log at `DEBUG` level only when the exception was a `UrlError`).

I'm fine with dropping the `Unhandled exception` log here.